File: js/10.js
// Summation of primes
   
// Problem 10

// The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

// Find the sum of all the primes below two million.

let primes = [2,3,5,7];

let sum = 2+3+5+7;

let current = 9;

while(current < 2000000){
    current += 2;
    let divisibleByAny = false;
    for (let i = 0; i < primes.length; i++){
        if (current % primes[i] == 0){
            divisibleByAny = true;
            break;
        }
    }
    if (!divisibleByAny){
        primes.push(current);
        sum += current;
    }
}

console.log(sum); // Answer: 142913828922

File: python/10.py
# Summation of primes
   
# Problem 10

# The sum of the primes below 10 is 2 + 3 + 5 + 7 = 17.

# Find the sum of all the primes below two million.

primes = [2]

def is_prime(n):
    for i in primes:
        if n % i == 0:
            return False
    return True

i = 1

while i+2 < 2000000:
    i += 2
    if is_prime(i):
        primes.append(i)

print(sum(primes))